components/director/internal/domain/ordvendor/service.go

```diff
@@ -45,27 +49,23 @@ func NewService(vendorRepo VendorRepository, uidService UIDService) *service {
 }
 
 // Create creates a new Vendor.
-func (s *service) Create(ctx context.Context, applicationID string, in model.VendorInput) (string, error) {
-	tnt, err := tenant.LoadFromContext(ctx)
-	if err != nil {
-		return "", err
-	}
-
+func (s *service) Create(ctx context.Context, resourceType resource.Type, resourceID string, in model.VendorInput) (string, error) {
 	id := s.uidService.Generate()
-	vendor := in.ToVendor(id, &applicationID)
+	vendor := in.ToVendor(id, resourceType, resourceID)
 
-	if err = s.vendorRepo.Create(ctx, tnt, vendor); err != nil {
-		return "", errors.Wrapf(err, "error occurred while creating a Vendor with id %s and title %s for Application with id %s", id, vendor.Title, applicationID)
+	if err := s.createVendor(ctx, vendor, resourceType); err != nil {
+		return "", errors.Wrapf(err, "error occurred while creating a Vendor with id %s and title %s", id, vendor.Title)
 	}
-	log.C(ctx).Debugf("Successfully created a Vendor with id %s and title %s for Application with id %s", id, vendor.Title, applicationID)
+
+	log.C(ctx).Debugf("Successfully created a Vendor with id %s and title %s", id, vendor.Title)
 
 	return vendor.OrdID, nil
 }
 
 // CreateGlobal creates a new Global Vendor (with NULL app_id).
 func (s *service) CreateGlobal(ctx context.Context, in model.VendorInput) (string, error) {
 	id := s.uidService.Generate()
-	vendor := in.ToVendor(id, nil)
+	vendor := in.ToVendor(id, "", "")
```

Comment:
If both parameters are `""` here i think that will be created a vendor with `application id `equal to "". I think we rather want nil as a value

Reply:
Nope, there won't be neither AppTemplateVersion, nor App 
<img width="807" alt="image" src="https://github.com/kyma-incubator/compass/assets/17293744/535a641b-ccb7-4fc8-99f8-e279fdb4508d">